media-tv/tvbrowser-bin/tvbrowser-bin-4.2.1.ebuild

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit desktop
+
+DESCRIPTION="Themeable and easy to use TV Guide - written in Java"
+HOMEPAGE="https://www.tvbrowser.org"
+MY_PN="tvbrowser"
+SRC_URI="mirror://sourceforge/project/tvbrowser/TV-Browser%20Releases%20%28Java%2011%20and%20higher%29/${PV}/${MY_PN}_${PV}_bin.tar.gz -> ${P}.tar.gz"
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+
+JAVA_SLOT=11
+RDEPEND="|| ( dev-java/openjdk:${JAVA_SLOT}
+	dev-java/openjdk-bin:${JAVA_SLOT}
+	dev-java/openjdk-jre-bin:${JAVA_SLOT} )"
+
+S="${WORKDIR}"/"${MY_PN}-${PV}"
+
+src_install() {
+	# Copy files and directories
+	insinto /opt/"${P}"
+	doins -r *
+
+	# Generate launcher
+	exeinto /opt/bin
+	newexe - tvbrowser <<-_EOF_
+	#!/bin/bash
+	export JAVA_OPTS="\${JAVA_OPTS} -Dpropertiesfile=/opt/${P}/linux.properties"
+	for JAVA_DIR in openjdk-jre-bin openjdk-bin openjdk; do
+		if [ -e /opt/\${JAVA_DIR}-${JAVA_SLOT}/bin ]; then
```

Comment:
there's no dir for openjdk:11 in `/opt`
it's in `"/usr/$(get_libdir)/openjdk-${JAVA_SLOT}"`

Reply:
I see. Then, where is the bin dir of openjdk:11? /usr/$(get_libdir)/openjdk-11/bin? Because the script needs to pass this to PATH.

Reply:
I think there should be better way than altering PATH.
I think you should set `JAVA_PROGRAM_DIR` before launching the script.
note, that you need trailing slash in that variable.

```bash
# locate and test the java executable
if [ "$JAVA_PROGRAM_DIR" == "" ]; then
  if ! command -v java &>/dev/null; then
    echo $MSG10
    if ! look_for_java ; then
      exit 1
    fi
  else
    echo $MSG11
    if ! check_version ; then
      if ! look_for_java ; then
        exit 1
      fi
    fi
  fi
fi
```

here's what the script does, if `JAVA_PROGRAM_DIR` is exported it will just use it without any checks. but make sure it's commented out in the launcher script.

all you have to do is properly export JAVA_PROGRAM_DIR

you can't use `$(get_libdir)` in a script outside portage, but you can just use a wildcard


something like this:

```bash
# create array of possible java installations
POSSIBLE_JAVA_PROGRAM_DIRS=$(
	"${EPREFIX}/usr/lib*/openjdk-${JAVA_SLOT}"
	"${EPREFIX}/opt/openjdk-bin"
	"${EPREFIX}/opt/openjdk-jre-bin"
)
#note that '${POSSIBLE_JAVA_PROGRAM_DIRS[@]}' HAS to be unquoted below
for POSSIBLE_DIR in ${POSSIBLE_JAVA_PROGRAM_DIRS[@]}; do
		if [[ -x  ${JAVA_DIR/}bin/java ]]; then
				export JAVA_PROGRAM_DIR="${POSSIBLE_DIR}/bin/"
				break
		fi
done
```

source version HAS to be first, don't try to force JRE first, JRE should be checked last.

also make sure that launcher script that ships with the app has JAVA_PROGRAM_DIR commented out!

Reply:
and please actually test it with all implementations.

Reply:
1.  I assume EPREFIX is only available in the ebuild, so I cannot use it in the script. I believe the wrapper script should be able to choose between the available openjdk packages at *runtime*, because the user could later decide to switch packages and would then need to recompile if this is handled in the ebuild.
2. I could try to somehow propagate the java slot to the wrapper script, using sed. The same with the ${P}. What would be a good way to do so? Copy the wrapper script from ${FILES_DIR} somewhere, run sed and then doins?
3. I have now compiled all three openjdk packages and will test them.